bikesharingweb/components/ErrorPanel.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react'
+
+export default class ErrorPanel extends Component {
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        return (
+            <div>
+                {this.props.errorMessage != null &&
```

Comment:
>props [](start = 22, length = 5)

Shouldn't this be `this.state`?  IIRC 'state' vs 'props' is a subtle but important difference.

Reply:
I'm not an expert tho so I could be wrong :)

---
In reply to: [266672761](https://github.com/ContosoBikeRental/BikeSharingSampleApp/pull/21#discussion_r266672761) [](ancestors = 266672761)

Reply:
I believe that using props here is the right way to do it, as the parent is providing this value as a property of the ErrorPanel component:
<ErrorPanel errorMessage={this.state.errorMessage} />
The state could be used to store and render the value, but I'm not sure of how it would retrieve the value from the parent in a standard way.

---
In reply to: [266672852](https://github.com/ContosoBikeRental/BikeSharingSampleApp/pull/21#discussion_r266672852) [](ancestors = 266672852,266672761)